mvvm-pattern/README.md

```diff
@@ -1,3 +1,270 @@
 # MVVM-Pattern
 
-[Zur Übersicht](../README.md)
\ No newline at end of file
+[Zur Übersicht](../README.md)
+
+<h2>Was ist MVVM?</h2>
+Das MVVM-Pattern (Model View ViewModel Muster) wurde erstmals 2005 von Microsoft entwickelt und veröffentlicht und sollte ursprünglich nur in WPF-Anwendungen eingesetzt werden. Inzwischen wird es, ähnlich wie MVC, in anderen Bereichen ebenfalls eingesetzt.
+
+MVVM ist ein Entwurfsmuster und eine Variante des MVC-Musters (Model-View-Controller). Das MVVM-Muster dient zur Trennung der Logik von der grafischen Benutzeroberfläche und findet Einsatz in WPF, JavaFX, HTML5 und Xamarin.Forms. 
+
+<h2>Verwendung</h2>
+MVVM enthält folgende drei Komponenten:
+
+* Model: Das Model ist die Datenzugriffsschicht auf die Inhalte, die der Benutzer manipulieren kann. Das Model benachritigt UI-Elemente wenn Daten geändert worden sind und führt Validierungen durch. Hier ist auch die gesammte Geschäftslogik enthalten.
+* View: Die View ist das was der Benutzer im Programm sieht. Sie besteht aus allen grafischen Elementen die das Programm ausmachen. Die View bindet sich an Eigenschaften aus dem ViewModel um Inhalte darzustellen, zu manipulieren oder um Benutzereingaben weiterzuleiten. 
+* ViewModel: Das ViewModel entählt die Logik der UI und ist das Bindeglied zwischen Model und View. Das ViewModel kann auf der einen Seite auf Methoden und Dienste des Models zugreifen und diese ausführen. Auf der anderen Seite stellt es öffentliche (public) Eigenschaften und Befehle (Commands) zur Verfügung, an welche sich die View binden kann. Das ViewModel darf jedoch keine Kenntnis über die View haben.
+
+Nachdem für die Verwendung des MVVM-Patterns die sogenannten Datenbindung nötig ist (für die Bindung der Properties und Commands an UI-Elemente), kann MVVM nur in jenen Sprachen verwendet werden, welche einen solchen Binder-Mechanismus zu Verfügung stellen. Für .NET und JavaFX werden hierfür Views aus XAML-Markup bzw. FXML erstellt und anschließend das Databinding implementiert. Die Logik hinter der UI wird danach mit C# oder Java implementiert.
+
+Seit 2010 gibt es MVVM auch für JavaScript bzw. HTML. Hierfür wurde der Datenbindungsmechanismus portiert, die daraus entstanden JavaScript-Bibliothek heit Knockout.js. Mittlerweile findet MVVM auch Anwendung in anderen JavaScript-Frameworks wie zum Beispiel AngularJS. 
+
+MVVM kann auch für die Entwicklung mobiler Apps verwendet werden. Für Android und iOS kann mit Hilfe von Xamarin.Forms ebenfalls das MVVM-Pattern implementiert werden. Die Gestaltung der UI erfolgt wie in WPF durch XAML-Markup, die Logik wird in C# geschrieben.
+
+<h2>Genauere Betrachtung</h2>
+<h3>Das Model</h3>
+
+Das Model stellt eine Abbildung der Daten dar, welche für die visuelle Anwendung benötigt werden. Es enthält jene Daten, welche dem Benutzer zur Verfügung gestellt und von ihm manipuliert werden. 
+
+Das Model stellt folgende Funktionalitäten bereit:
```

Comment:
Zusätzlich vielleicht noch was bzgl. Geschäftslogik (Persistierung von neuen Datensätzen in DB; Aufruf von Aktionen in anderen Systemen -> zB. Warenwirtschaftssystem, etc.) aufnehmen, welche durch Commandos im ViewModel getriggert werden kann.